src/core/Basics/Song.cpp

```diff
@@ -725,6 +726,53 @@ Song* SongReader::readSong( const QString& sFileName )
 	pSong->setPlaybackTrackEnabled( bPlaybackTrackEnabled );
 	pSong->setPlaybackTrackVolume( fPlaybackTrackVolume );
 	pSong->setActionMode( actionMode );
+	
+	// pan law
+	Sampler* pSampler = AudioEngine::get_instance()->get_sampler();
+	QString sPanLawType( LocalFileMng::readXmlString( songNode, "pan_law", "RATIO_STRAIGHT_POLYGONAL" ) );
+	if ( sPanLawType == "RATIO_STRAIGHT_POLYGONAL" ) {
+		pSampler->setPanLawType( pSampler->RATIO_STRAIGHT_POLYGONAL );
+	} else if ( sPanLawType == "RATIO_CONST_POWER" ) {
+		pSampler->setPanLawType( pSampler->RATIO_CONST_POWER );
+	} else if ( sPanLawType == "RATIO_CONST_SUM" ) {
+		pSampler->setPanLawType( pSampler->RATIO_CONST_SUM );
+	} else if ( sPanLawType == "LINEAR_STRAIGHT_POLYGONAL" ) {
+		pSampler->setPanLawType( pSampler->LINEAR_STRAIGHT_POLYGONAL );
+	} else if ( sPanLawType == "LINEAR_CONST_POWER" ) {
+		pSampler->setPanLawType( pSampler->LINEAR_CONST_POWER );
+	} else if ( sPanLawType == "LINEAR_CONST_SUM" ) {
+		pSampler->setPanLawType( pSampler->LINEAR_CONST_SUM );
+	} else if ( sPanLawType == "POLAR_STRAIGHT_POLYGONAL" ) {
+		pSampler->setPanLawType( pSampler->POLAR_STRAIGHT_POLYGONAL );
+	} else if ( sPanLawType == "POLAR_CONST_POWER" ) {
+		pSampler->setPanLawType( pSampler->POLAR_CONST_POWER );
+	} else if ( sPanLawType == "POLAR_CONST_SUM" ) {
+		pSampler->setPanLawType( pSampler->POLAR_CONST_SUM );
+	} else if ( sPanLawType == "QUADRATIC_STRAIGHT_POLYGONAL" ) {
+		pSampler->setPanLawType( pSampler->QUADRATIC_STRAIGHT_POLYGONAL );
+	} else if ( sPanLawType == "QUADRATIC_CONST_POWER" ) {
+		pSampler->setPanLawType( pSampler->QUADRATIC_CONST_POWER );
+	} else if ( sPanLawType == "QUADRATIC_CONST_SUM" ) {
+		pSampler->setPanLawType( pSampler->QUADRATIC_CONST_SUM );
+	} else if ( sPanLawType == "LINEAR_CONST_K_NORM" ) {
+		pSampler->setPanLawType( pSampler->LINEAR_CONST_K_NORM );
+	} else if ( sPanLawType == "POLAR_CONST_K_NORM" ) {
+		pSampler->setPanLawType( pSampler->POLAR_CONST_K_NORM );
+	} else if ( sPanLawType == "RATIO_CONST_K_NORM" ) {
+		pSampler->setPanLawType( pSampler->RATIO_CONST_K_NORM );
+	} else if ( sPanLawType == "QUADRATIC_CONST_K_NORM" ) {
+		pSampler->setPanLawType( pSampler->QUADRATIC_CONST_K_NORM );
+	} else {
+		pSampler->setPanLawType( pSampler->RATIO_STRAIGHT_POLYGONAL );
+		WARNINGLOG( "Unknown pan law type in import song. Set default." );
+	}
+
+	float fPanLawKNorm = LocalFileMng::readXmlFloat( songNode, "pan_law_k_norm", K_NORM_DEFAULT );
+	if ( fPanLawKNorm <= 0.0 ) {
+		fPanLawKNorm = K_NORM_DEFAULT;
```

Comment:
Could you make `K_NORM_DEFAULT` a static member variable of `Sampler`? This way it will be perfectly clear where to find its definition and documentation.

Reply:
ok